pennylane/_qubit_device.py

```diff
@@ -401,6 +406,129 @@ def execute_new(self, circuit, **kwargs):
         #     self.tracker.record()
         return results
 
+    def shot_vec_statistics(self, circuit):
+        """Process measurement results from circuit execution using a device
+        with a shot vector and return statistics.
+
+        This is an auxiliary method of execute_new and uses statistics_new.
+
+        When using shot vectors, measurement results for each item of the shot
+        vector are contained in a tuple.
+
+        Args:
+            circuit (~.tapes.QuantumTape): circuit to execute on the device
+
+        Raises:
+            QuantumFunctionError: if the value of :attr:`~.Observable.return_type` is not supported
+
+        Returns:
+            tuple: stastics for each shot item from the shot vector
+        """
+        results = []
+        s1 = 0
+
+        ret_types = [m.return_type for m in circuit.measurements]
+        counts_exist = any(ret is qml.measurements.Counts for ret in ret_types)
+        single_measurement = len(circuit.measurements) == 1
+
+        for shot_tuple in self._shot_vector:
+            s2 = s1 + np.prod(shot_tuple)
+            r = self.statistics_new(
+                circuit.observables, shot_range=[s1, s2], bin_size=shot_tuple.shots
+            )
+
+            # This will likely be required:
+            # if qml.math._multi_dispatch(r) == "jax":  # pylint: disable=protected-access
+            #     r = r[0]
+
+            if single_measurement:
+                r = r[0] if counts_exist else qml.math.squeeze(r)
```

Comment:
Why are we using squeezing here? Should it not be already good as you changed `statistics_new`

Reply:
The prior changes didn't affect squeezing. Made new changes and added an explanatory comment.